tests/test_gather_stock_time_series.py

```diff
@@ -0,0 +1,88 @@
+import pyntrinio
+from datetime import datetime
+
+# helper data
+api_key = 'OjhlMjhjNTBmY2IyMWJiMWE0MTExYjQwNWZmZTVkZWM1'
+ticker = 'AAPL'
+start_date = '2005-03-01' 
+end_date = '2020-01-15' 
+
+# test that return type is dictionary by default
+def test_default_output_type():
+    """
+    Test that return type is dictionary by default
+    """
+    assert type(gather_stock_time_series(api_key, ticker)) == 'dict'
+
+# test that return type is a pandas dataframe when specified
+def test_pddf_output_type():
+    """
+    Test that return type is a pandas dataframe when specified
+    """
+    assert type(gather_stock_time_series(api_key, ticker, output_format='pddf')) == 'pandas.core.frame.DataFrame'
+
+# test that you get an error when you put in an incorrect API key
+def api_key_test():
+    """
+    Give a wrong api_key, see if the function correctly handles the exception.
+    """
+    msg = "Incorrect API Key - please input a valid API key as a string"
+    assert gather_stock_time_series('wrong api key!!', ticker) == msg
+
+# test that you get an error when you put in an invalid date format
+def date_format_test():
+    """
+    See if the function correctly handles the input dates in wrong formats.
+    """
+    msg = "Invalid Date format - please input the date as a string with format %Y-%m-%d"
+    assert gather_stock_time_series(api_key, ticker, '2018-15-12') == msg
+
+# test that you get an error when the end date is before the start date
+def test_start_date_prior_to_end_date():
+    """
+    Test that you get an error when the end date is before the start date
+    """
+    msg = "Please choose a start date that is before the end date"
+    assert type(gather_stock_time_series(api_key, ticker, start_date=end_date, end_date=start_date)) == msg
```

Comment:
I can't find the corresponding design in the function file @_@

Reply:
Same issue as other file, I added the wrong version - fixed in new commit